app/models/manageiq/providers/vmware/infra_manager/provision/configuration/disk.rb

```diff
@@ -21,13 +21,38 @@ def build_config_disk_spec(vmcs)
     end
   end
 
-  def get_scsi_controller_info
-    inventory_hash = source.with_provider_connection do |vim|
-      vim.virtualMachineByMor(source.ems_ref_obj)
+  def build_disk_relocate_spec(datastore)
+    VimArray.new('ArrayOfVirtualMachineRelocateSpecDiskLocator') do |relocate_spec_array|
+      (disks + get_new_disks).map do |disk|
```

Comment:
@d-m-u does the response of `get_new_disks` have a `.key` ?

I also wonder if we even need a RelocateSpec for a new disk, it isn't being relocated and we could just set eagerlyScrub when we build the backing